damnit/ctxsupport/ctxrunner.py

```diff
@@ -45,6 +46,45 @@ class DataType(Enum):
     Timestamp = "timestamp"
 
 
+def get_run_info(run, run_no, proposal):
+    import yaml
+    import requests
```

Comment:
I would guess James has done this because the `ctxrunner` file is imported in DAMNIT's own frontend & backend processes, as well as the subprocesses executing the context file. By hiding the imports inside functions like this, DAMNIT doesn't require them.

We increasingly run the context file in a separate Python environment from the DAMNIT application - perhaps it's time to make that the default.

Reply:
Yeah, though with the new class in 8ce6281 it's a little annoying to import everything where necessary so I moved them to the top.